### PR TITLE
Test, build and CI improvements and fixes

### PR DESCRIPTION
**Pull request by guillemj**
This is an assorted bag of fixes and improvements for the build and test infrastructure.

Cc: @ammarfaizi2

### CODE REVIEW

File: test/empty-eownerdead.c [hunk 1]
```diff
@@ -32,7 +32,10 @@ int main(int argc, char *argv[])
 	if (ret < 0) {
 		int __e = errno;
 
-		fprintf(stderr, "child: sqe submit failed: %s\n", strerror(__e));
+		if (__e == EOWNERDEAD)
+			fprintf(stderr, "sqe submit unexpected failure due old kernel bug: %s\n", strerror(__e));
+		else
+			fprintf(stderr, "sqe submit unexpected failure: %s\n", strerror(__e));
 		goto err;
 	}
```
Comment (ammarfaizi2) [inline test/empty-eownerdead.c:40]: Ahh good, it should be more acceptable than 84875f42cab07d4b39d64b043cf9dbbdfe618674 ("test/empty-eownerdead: Only expect EOWNERDEAD as test success").

**Review by ammarfaizi2 — APPROVED** [review]
Looks good, reviewed and tested by me.

File: test/empty-eownerdead.c (hunk 1 above)
Comment (ammarfaizi2) [reply test/empty-eownerdead.c:40]: > Ahh good, it should be more acceptable than 84875f42cab07d4b39d64b043cf9dbbdfe618674 ("test/empty-eownerdead: Only expect EOWNERDEAD as test success").

Oops, wrong reference.
6e72065e39e6bb7e744c6ba24d1a62a8d84d3e27 ("test/empty-eownerdead: Only expect EOWNERDEAD as test success").

**Comment by ammarfaizi2** [comment]
Hi @guillemj,

Sorry, it may be too late, but I missed something when reviewing this. I think `-g -O2 -Wall -Wextra` should be mandatory flags. But when we use `CFLAGS=-Werror`, they don't get appended.

---

## Related Code
https://github.com/axboe/liburing/blob/b7c779a90a987f7e65405c7985c52212fcf3cb53/src/Makefile#L9-L10

https://github.com/axboe/liburing/blob/b7c779a90a987f7e65405c7985c52212fcf3cb53/test/Makefile#L13-L26

---

## Without `CFLAGS`, we have `-g -O2 -Wall -Wextra` here (it's fine).
```
make V=1

gcc -D_GNU_SOURCE -D__SANE_USERSPACE_TYPES__ -I../src/include/ -include ../config-host.h -g -O2 -Wall -Wextra -Wno-unused-parameter -Wno-sign-compare -Wstringop-overflow=0 -Warray-bounds=0 -o accept accept.c helpers.o -L../src/ -luring
gcc -D_GNU_SOURCE -D__SANE_USERSPACE_TYPES__ -I../src/include/ -include ../config-host.h -g -O2 -Wall -Wextra -Wno-unused-parameter -Wno-sign-compare -Wstringop-overflow=0 -Warray-bounds=0 -o accept-link accept-link.c helpers.o -L../src/ -luring -lpthread
gcc -D_GNU_SOURCE -D__SANE_USERSPACE_TYPES__ -I../src/include/ -include ../config-host.h -g -O2 -Wall -Wextra -Wno-unused-parameter -Wno-sign-compare -Wstringop-overflow=0 -Warray-bounds=0 -o accept-reuse accept-reuse.c helpers.o -L../src/ -luring
gcc -D_GNU_SOURCE -D__SANE_USERSPACE_TYPES__ -I../src/include/ -include ../config-host.h -g -O2 -Wall -Wextra -Wno-unused-parameter -Wno-sign-compare -Wstringop-overflow=0 -Warray-bounds=0 -o accept-test accept-test.c helpers.o -L../src/ -luring
```

---


## With `CFLAGS`, we don't have `-g -O2 -Wall -Wextra` here (it's not fine).

I think we should have them, even when `CFLAGS` is specified.

```
make V=1 CFLAGS=-Werror

gcc -D_GNU_SOURCE -D__SANE_USERSPACE_TYPES__ -I../src/include/ -include ../config-host.h -Werror -Wno-unused-parameter -Wno-sign-compare -Wstringop-overflow=0 -Warray-bounds=0 -o helpers.o -c helpers.c
gcc -D_GNU_SOURCE -D__SANE_USERSPACE_TYPES__ -I../src/include/ -include ../config-host.h -Werror -Wno-unused-parameter -Wno-sign-compare -Wstringop-overflow=0 -Warray-bounds=0 -o 232c93d07b74-test 232c93d07b74-test.c helpers.o -L../src/ -luring -lpthread
gcc -D_GNU_SOURCE -D__SANE_USERSPACE_TYPES__ -I../src/include/ -include ../config-host.h -Werror -Wno-unused-parameter -Wno-sign-compare -Wstringop-overflow=0 -Warray-bounds=0 -o 35fa71a030ca-test 35fa71a030ca-test.c helpers.o -L../src/ -luring -lpthread
gcc -D_GNU_SOURCE -D__SANE_USERSPACE_TYPES__ -I../src/include/ -include ../config-host.h -Werror -Wno-unused-parameter -Wno-sign-compare -Wstringop-overflow=0 -Warray-bounds=0 -o 500f9fbadef8-test 500f9fbadef8-test.c helpers.o -L../src/ -luring
gcc -D_GNU_SOURCE -D__SANE_USERSPACE_TYPES__ -I../src/include/ -include ../config-host.h -Werror -Wno-unused-parameter -Wno-sign-compare -Wstringop-overflow=0 -Warray-bounds=0 -o 7ad0e4b2f83c-test 7ad0e4b2f83c-test.c helpers.o -L../src/ -luring
```

**Comment by guillemj** [comment]
> Sorry, it may be too late, but I missed something when reviewing this. I think `-g -O2 -Wall -Wextra` should be mandatory flags. But when we use `CFLAGS=-Werror` from command, they don't get appended.

Ah right, missed adapting the GitHub Actions. I don't think all of these should be mandatory TBH, The -g and -O2 seem clearly optional, as you might want to build w/o debugging symbols or w/o optimizations, for debugging purposes, and while later options should override earlier ones, it seems unclean to me to include both. :) Contrary to the various -Wno-<something> which seem required as are dependent on the current code, and would otherwise emit warnings, the -Wall -Wextra seem optional to me, and potentially problematic if the user passes -Werror when using some other compiler, or a newer or older compiler version, which might be part of global compiler flags setting and might need special handling for this specific package.

Will send a MR to fix that.
